frontend/public/components/routes.tsx

```diff
@@ -186,13 +188,47 @@ const calcTrafficPercentage = (weight: number, route: any) => {
   return `${percentage.toFixed(1)}%`;
 };
 
+const getIngressStatusForHost = (hostname, ingresses) => {
+  return _.find(ingresses, {host: hostname});
+};
+
+const showCustomRouteHelp = (ingress, annotations) => {
+  if (!ingress || !_.some(ingress.conditions, {type: 'Admitted', status: 'True'})) {
+    return false;
+  }
+
+  if (_.get(annotations, 'openshift.io/host.generated') === 'true') {
+    return false;
+  }
+
+  if (!ingress.host || !ingress.routerCanonicalHostname) {
+    return false;
+  }
+
+  return true;
+};
+
 const RouteTargetRow = ({route, target}) => <tr>
   <td><ResourceLink kind={target.kind} name={target.name} namespace={route.metadata.namespace} title={target.name} /></td>
   <td>{target.weight}</td>
   <td>{calcTrafficPercentage(target.weight, route)}</td>
 </tr>;
 
-const RouteIngressStatus: React.SFC<RouteIngressStatusProps> = ({ingresses}) =>
+const CustomRouteHelp = ({host, canonicalHostname, id, learnMoreLink=null}) => <Popover id={id}
```

Comment:
Why is `id` needed?

Reply:
This prop should be `routerCanonicalHostname`. It needs to be clear that it's the router hostname, not the route hostname.

Reply:
Please add a `CustomRouteHelpProps` type

Reply:
id was not needed.